spec/services/messages/channels_spec.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Messages
+  describe Channels do
+    let(:offer) { create(:offer) }
+    let(:user) { create(:user) }
+    let(:channel) { Messages::Channels.new(current_user: user, is_private: true, app_name: DONOR_APP, messageable: offer) }
+
+    describe '#initialize' do
+      it 'creates instance for variables' do
+        expect(channel.current_user).to eq(user)
+        expect(channel.messageable).to eq(offer)
+        expect(channel.app_name).to eq(DONOR_APP)
+        expect(channel.is_private).to be_truthy
+      end
+    end
+
+    describe '#related_users' do
+      let!(:supervisor) { create(:user, :supervisor) }
+      let!(:supervisor2) { create(:user, :supervisor) }
+      let!(:reviewer) { create(:user, :reviewer) }
+
+      context 'when ADMIN and DONOR app' do
+        let!(:donor) { create(:user) }
+        let(:offer) { create(:offer, created_by: donor) }
+
+        before do
+          create_list(:user, 2, disabled: true)
+          create_list(:user, 3, :order_fulfilment)
+          create_list(:user, 3, :charity)
+        end
+
+        context 'in DONOR channel from DONOR app' do
+          it 'fetches users with REVIEWER or SUPERVISOR role' do
+            result = Messages::Channels.new(current_user: donor, is_private: false, app_name: DONOR_APP, messageable: offer).related_users
+            expected = [[User.supervisors.map(&:id), User.reviewers.map(&:id)].flatten].flatten.map { |id| { :id => id, :name => User.find(id).full_name} }
+            expect(result).to match_array(expected)
+          end
+
+          it 'does not fetch the current user' do
+            result = Messages::Channels.new(current_user: donor, is_private: false, app_name: DONOR_APP, messageable: offer).related_users
+            expected = { :id => donor.id, :name => donor.full_name }
+            expect(result).not_to include(expected)
+          end
+
+          it 'does not fetch any stock or browse related users' do
+            result = Messages::Channels.new(current_user: donor, is_private: false, app_name: DONOR_APP, messageable: offer).related_users
+
+            expected = [[User.order_administrator.map(&:id), User.order_fulfilment.map(&:id)].flatten].flatten.map { |id| {:id => id, :name => User.find(id).full_name} }
```

Comment:
```suggestion
            not_expected = [[User.order_administrator.map(&:id), User.order_fulfilment.map(&:id)].flatten].flatten.map { |id| {:id => id, :name => User.find(id).full_name} }
```

also need to include charity user in the exclusion list here